semgrep_metrics.atd

```diff
@@ -156,6 +156,33 @@ type pro_features = {
     ?diffDepth <ocaml mutable>: int option;
 }
 
+type analysis_type <ocaml attr="deriving show"> = [ 
+    | Intraprocedural 
+    | Interprocedural 
+    | Interfile
+]
+
+type secrets_config 
+     <ocaml attr="deriving show"> = {
+  permitted_origins: string;
+}
+
+type pro_config 
+     <ocaml attr="deriving show"> = {
+  analysis_type: analysis_type;
+  ?secrets_config: secrets_config option;
+  pro_langs: bool;
+}
+
+(* Since v1.54.0 *)
+type engine_config
+     <ocaml attr="deriving show">
+     <python decorator="dataclass(frozen=True)"> = [
+  | OSS
+  | PRO of pro_config
+]
```

Comment:
In the world where we just have `product_config`, I wouldn't have this `engine_config` idea. 

I think this PR really reveals the existing problems with the paradigm of "engineRequested", which made sense when we only had one product where this was relevant but now feels weird. It feels strange to me to have the secrets configuration be under the `pro_config` and muddles even further what "pro" means.

Reply:
We could just make `engine_config` be `pro_config option` (and rename somewhat) if there are objections to calling this "pro_xxx".